HomeWorks/HW3/LZW/BWT.cs

```diff
@@ -0,0 +1,90 @@
+namespace BWT;
+
+public static class BurrowsWheelerTransform
+{
+    public static (string transformed, int index) Transform(string input)
+    {
+        if (!input.EndsWith("$"))
+        {
+            input += '$'; // Добавляем уникальный терминатор
+        }
+
+        int n = input.Length;
+        int[] indices = Enumerable.Range(0, n).ToArray(); // Создаём массив индексов
+
+        // Сортируем индексы, используя кастомный компаратор
+        Array.Sort(indices, (i, j) =>
+        {
+            for (int k = 0; k < n; k++)
+            {
+                char ci = input[(i + k) % n];
+                char cj = input[(j + k) % n];
+                if (ci != cj)
+                {
+                    return ci - cj;
+                }
+            }
+
+            return 0;
+        });
+
+        // Формируем BWT-строку и находим позицию исходной строки
+        char[] lastColumn = new char[n];
```

Comment:
```suggestion
        var lastColumn = new char[n];
```
тут и ниже, не пишите один и тот же таип дважды